### PR TITLE
Pass LeaderElectionNamespace when not in k8s

### LINKED ISSUE #382: `LeaderElectionNamespace` is required when run controller locally
`controller-runtime` cannot check the namespace when out-of-cluster, and complains below error when run function-mesh locally:

```
ubuntu@instance-20211013-2148:~/function-mesh$ make run
/home/ubuntu/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
/home/ubuntu/go/bin/controller-gen "crd:maxDescLen=0,trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
go run ./main.go
2022-06-06T16:38:05.441+0800    ERROR   setup   unable to start manager {"error": "unable to find leader election namespace: not running in-cluster, please specify LeaderElectionNamespace"}
github.com/go-logr/zapr.(*zapLogger).Error
        /home/ubuntu/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
main.main
        /home/ubuntu/function-mesh/main.go:106
runtime.main
        /usr/local/go/src/runtime/proc.go:250
exit status 1
make: *** [Makefile:62: run] Error 1
```

related codes are:
https://github.com/kubernetes-sigs/controller-runtime/blob/release-0.6/pkg/leaderelection/leader_election.go#L95-L111

temporary fix:

provides a `LeaderElectionNamespace` or disable `LeaderElection`

```diff
        mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-               Scheme:                 scheme,
-               MetricsBindAddress:     metricsAddr,
-               HealthProbeBindAddress: healthProbeAddr,
-               Port:                   9443,
-               LeaderElection:         enableLeaderElection,
-               LeaderElectionID:       leaderElectionID,
-               Namespace:              namespace,
-               CertDir:                certDir,
+               Scheme:                  scheme,
+               MetricsBindAddress:      metricsAddr,
+               HealthProbeBindAddress:  healthProbeAddr,
+               Port:                    9443,
+               LeaderElection:          enableLeaderElection,
+               LeaderElectionNamespace: "default",
+               LeaderElectionID:        leaderElectionID,
+               Namespace:               namespace,
+               CertDir:                 certDir,
        })
```

### PR DESCRIPTION
**Pull request by jiangpengcheng**
Fixes #382 

### Motivation

`LeaderElectionNamespace` is required when operator is running outside of k8s

### Modifications

get `LeaderElectionNamespace` from env `LEADER_ELECTION_NAMESPACE` or use `default`

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by freeznet — COMMENTED** [review]
Is it possible to make the `LeaderElectionNamespace` configurable as a parameter?

**Comment by jiangpengcheng** [comment]
updated

**Review by freeznet — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -57,7 +57,7 @@ manager: generate fmt vet
 
 # Run against the configured Kubernetes cluster in ~/.kube/config
 run: generate fmt vet manifests
-	go run ./main.go
+	go run ./main.go --leader-election-namespace default
```
[the remark this answers is not in the document]
Comment (jiangpengcheng) [reply Makefile]: good idea

File: main.go [hunk 2]
```diff
@@ -59,6 +60,8 @@ func main() {
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&leaderElectionID, "leader-election-id", "a3f45fce.functionmesh.io",
 		"the name of the configmap that leader election will use for holding the leader lock.")
+	flag.StringVar(&leaderElectionNamespace, "leader-election-namespace", "",
+		"the namespace in which the leader election configmap will be created")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true,
```
Comment (tpiperatgod) [inline main.go:65]: Can we set the `enable-leader-election` to false by default?

**Review by tpiperatgod — APPROVED** [review]
lgtm

File: main.go (hunk 2 above)
Comment (freeznet) [reply main.go:65]: +1

Comment (jiangpengcheng) [reply main.go:65]: then we need to set that to true manually in k8s cluster environment, is that ok

Comment (jiangpengcheng) [reply main.go:65]: seems already did that, I will update

**Pull request merged**